packages/commonwealth/client/scripts/views/components/Comments/CreateComment.tsx

```diff
@@ -158,55 +159,73 @@ export const CreateComment = ({
   }, [handleIsReplying, saveDraft, contentDelta]);
 
   return (
-    <div className="CreateComment">
-      <div className="attribution-row">
-        <div className="attribution-left-content">
-          <CWText type="caption">
-            {parentType === ContentType.Comment ? 'Reply as' : 'Comment as'}
-          </CWText>
-          <CWText
-            type="caption"
-            fontWeight="medium"
-            className={clsx('user-link-text', { disabled: !canComment })}
-          >
-            <User user={author} hideAvatar linkify />
-          </CWText>
-        </div>
-        {errorMsg && <CWValidationText message={errorMsg} status="failure" />}
-      </div>
-      <ReactQuillEditor
-        className="editor"
-        contentDelta={contentDelta}
-        setContentDelta={setContentDelta}
-        isDisabled={!canComment}
-        tooltipLabel="Join community to comment"
-      />
-      {tokenPostingThreshold && tokenPostingThreshold.gt(new BN(0)) && (
-        <CWText className="token-req-text">
-          Commenting in {activeTopic?.name} requires{' '}
-          {weiToTokens(tokenPostingThreshold.toString(), decimals)}{' '}
-          {app.chain.meta.default_symbol}.{' '}
-          {userBalance && (
-            <>
-              You have {weiToTokens(userBalance.toString(), decimals)}{' '}
-              {app.chain.meta.default_symbol}.
-            </>
-          )}
-        </CWText>
-      )}
-      <div className="form-bottom">
-        <div className="form-buttons">
-          {editorValue.length > 0 && (
-            <CWButton buttonType="tertiary" onClick={cancel} label="Cancel" />
-          )}
-          <CWButton
-            buttonWidth="wide"
-            disabled={disabled && !isAdmin}
-            onClick={handleSubmitComment}
-            label="Submit"
-          />
-        </div>
-      </div>
-    </div>
+    <>
+      { rootThread.archivedAt === null ? (
```

Comment:
Can we split this out into separate components, following the same pattern as in #4776, rather than using a big ternary? We can do an `if (archived) return ArchivedComponent; return MainComponent;` pattern to make this more readable.